### PR TITLE
Prepare for Angular 13.1. Move to ESM

### PR DESCRIPTION
**Pull request by SanderElias**
- chore(monorepo): finished NX migrate to 13. WIP, fixes in next commits
- fix(ng-lib): fix build errors from nx migration
- refactor(monorepo): partway done upgradiing to ng 13, still problems with ESM/CommonJS
- chore(monorepo): nX update
- fix(monorepo): upto ESM modules, PW is working, ppt and sps still broken
- chore(scully): still working on esm
- chore(scully): still ESM, but sps comes back to life, yet to figure out why results are incomplete.
- fix(monorepo) SPS now working, Fixed XhrFactory bug by providing my own
- fix(scully-plugin-puppeteer): fixed exit problem with ppt renderer
- fix(monorepo): make tests work with ESM
- chore(monorepo): pruge a few stale files from scully folder
- chore(monorepo): handle relative path differently bcs GA
- fix(monorepo): fix GA by adding mandatory node flags
- fix(monorepo): update workflow to cater ESM
- fix(monorepo): try less concurrency for guthub actions
- fix(scully): requireify the load of sentry
- fix(monorepo): use require for jsYaml
- fix(monorepo): replace jsYaml with yaml to fix GA build problem
- fix(ng-lib): move rx7 to peerdep to aid install failure in test
- ci(ng-lib): fix workflow to aid new setup
- fix(actions): fix typo in workflow
- fix(actions): fix workflow to install rx7
- fix(scully): fix versions in package.json for scully
- fix(monorepo): update workflow, and demo plugins to modules
- fix(actions): ok, now I'm going to try to use the force
- fix(actions): upgrade schematic and use the force on npm
- fix(scully): use require for path-to-regexp commonjs module
- fix(actions): update workflow to install develop version of ppt plugin && latest puppeteer
- fix(actions): move npm install as init schematics install wrong versions
- fix(actions): fix type in wrokflow
- fix(platform-server): make sure ready events doesn't fire premature
- chore(monorepo): update package-lock
- feat(monorepo): latest nx
- feat(actions): add support for next branch

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/scullyio/scully/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: .firebase/hosting.ZGlzdFxzdGF0aWNcZG9jLXNpdGVz.cache [hunk 1]
```diff
@@ -70,7 +70,7 @@ docs/learn/plugins/built-in-plugins/contentFolder/index.html,1596568073818,77c91
 docs/learn/plugins/built-in-plugins/json/index.html,1596568073463,55489660f9b8004978948fc26db370c6355bce308f9c995a73decd4c9b6e580f
 docs/learn/plugins/built-in-plugins/md/index.html,1596568073596,fa4befb3f162a18827a62a3acbd09df7858ba43bbde7502d549ceecbb3ec0628
 docs/learn/plugins/built-in-plugins/router-/index.html,1596568073823,e938b38eebcca70115c859042157300ca2da167a4e1b39a2ec042c73152ee969
-docs/learn/plugins/built-in-plugins/seoHrefOptimize/index.html,1596568073506,cdd031510ac8e5f37e2a4eaf72f7d29904688f70c8289aa0a38d4b9b7c33be39
+docs/learn/plugins/built-in-plugins/seoHrefOptimise/index.html,1596568073506,cdd031510ac8e5f37e2a4eaf72f7d29904688f70c8289aa0a38d4b9b7c33be39
```
Comment (gn-sky) [inline .firebase/hosting.ZGlzdFxzdGF0aWNcZG9jLXNpdGVz.cache:73]: interesting :) last time i checked optimi**Z**e was the right spelling

File: apps/sample-blog/src/app/app.component.ts [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
+        console.log(`
+        stable: ${stable}
+        AppComponent: ${this.currentState}
+        loc: ${this.loc.path(true)}
+        url: ${this.route.snapshot.pathFromRoot.map((x) => x.url).join('/')}
+        route: ${this.router.url},
+        ps: ${platformStrategy.constructor.name}
+        `);
+      })
+    ); //.subscribe();
```
Comment (gn-sky) [inline apps/sample-blog/src/app/app.component.ts:34]: do you want to keep this comment?

File: apps/sample-blog/src/app/app.sps.module.ts [hunk 3]
```diff
@@ -1,23 +1,17 @@
...
-
 @NgModule({
-  imports: [
-    BrowserModule.withServerTransition({ appId: 'serverApp' }),
-    AppModule,
-    ScullyPlatformServerModule,
-  ],
-  providers: [],
+  imports: [BrowserModule.withServerTransition({ appId: 'serverApp' }), AppModule, ScullyPlatformServerModule],
+  // providers: [{ provide: XhrFactory, useClass: ScullyXhrFactory }],
```
Comment (gn-sky) [inline apps/sample-blog/src/app/app.sps.module.ts:14]: do you want to keep this comment?

File: docs/Reference/plugins/built-in-plugins/seoHrefOptimize.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
 ---
-title: seoHrefOptimize Plugin
+title: seoHrefOptimise Plugin
```
Comment (gn-sky) [inline docs/Reference/plugins/built-in-plugins/seoHrefOptimize.md:2]: file name is still optimi**Z**e. Maybe do not change it to `s`?

File: jest.config.mjs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// export default
```
Comment (gn-sky) [inline jest.config.mjs:1]: `export default config;` in the end looks pretty explicit; remove comment maybe?

File: jest.preset.mjs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { createRequire } from 'module';
+
+// const require = createRequire(import.meta.url);
```
Comment (gn-sky) [inline jest.preset.mjs:3]: do you want to keep these comments?

File: apps/sample-blog/src/app/user/user.component.ts [hunk 7]
```diff
@@ -31,31 +23,24 @@ export class UserComponent implements OnInit {
   apiUser$ = this.userId$.pipe(
     switchMap((id) =>
       this.http.get<User>(`/api/users/${id}`).pipe(
-        catchError(() =>
-          of({
+        catchError((e) => {
```
Comment (gn-sky) [inline apps/sample-blog/src/app/user/user.component.ts:26]: if we'd like to log into console, let's do `console.error` for errors?

File: libs/scully/src/lib/fileHanderPlugins/markdown.ts [hunk 8]
```diff
@@ -1,25 +1,27 @@
...
-import 'prismjs/components/prism-markup';
-import 'prismjs/components/prism-markdown';
-import 'prismjs/components/prism-typescript';
-import 'prismjs/components/prism-jsx';
-import 'prismjs/components/prism-tsx';
-import 'prismjs/components/prism-docker';
+import Prism from 'prismjs';
+import loadLanguages from 'prismjs/components/index.js';
+loadLanguages(['javascript', 'typescript', 'css', 'scss', 'markdown', 'bash', 'json']);
+// import 'prismjs/components/prism-bash';
```
Comment (gn-sky) [inline libs/scully/src/lib/fileHanderPlugins/markdown.ts:11]: do you want to keep these comments?

Comment (SanderElias) [reply libs/scully/src/lib/fileHanderPlugins/markdown.ts:11]: udated

**Comment by SanderElias** [comment]
The current version is released to NPM under the `next` tag

**Comment by leoz2007** [comment]
> The current version is released to NPM under the `next` tag

Thanks for that!
How I can install Scully without schematics ?
`npm i @scullyio/scully@3.0.0-beta.0 `doesn't have any init

**Comment by SanderElias** [comment]
@leoz2007 A merge request isn't the best place to ask a question like this ;)
Npm I all the packacgs you need. (scully, ngLib, puppeteer, and init) with the @next flag,
Then start the the ngAdd schmatic like this:
```bash
npx ng g @scullyio/init:ng-add --renderer puppeteer
```
Or dint run the schematics, and manually add all the updates. There is an answer from me somewhere that explaines how to.

**Review by gn-sky — APPROVED** [review]

**Pull request merged**